use-shopping-cart/src/index.test.js

```diff
@@ -171,6 +171,45 @@ describe('useShoppingCart()', () => {
       expect(cart.current.cartCount).toBe(2)
       expect(cart.current.totalPrice).toBe(650)
     })
+
+    it('adds price_data from the metadata object from the 3rd param', () => {
+      const product = mockProduct()
+
+      act(() => {
+        cart.current.addItem(product, 1, {
+          type: 'tacos',
+          test: 'testing'
+        })
+      })
+
+      const entry = cart.current.cartDetails[product.id]
+
+      entry.price_data = {
+        type: 'tacos',
+        test: 'testing'
+      }
+
+      expect(cart.current.cartDetails).toStrictEqual({ [entry.id]: entry })
```

Comment:
This code doesn't seem to test any functionality. I tried removing this new feature and keeping these tests, which should break the tests bc the feature is no longer there, but the tests pass regardless.

You're comparing `cart.current.cartDetails` to an object with `cart.current.cartDetails[product.id]` in it. These are the same values so they'll always be true. I would suggest testing that the `price_data` property on the entry has the correct values you assigned.

Also, you overwrote the `price_data` property so it would always be true still. It's best not to tamper with the entry as it's what determines the result of our tests.